--- gratia-config-probe	2012-08-29 20:20:17.000000003 +0000
+++ gratia-config-probe.new	2013-02-26 20:59:45.000000003 +0000
@@ -173,7 +173,7 @@
   # specifically in the ProbeConfiguration.getFilenameFragment() function.  If
   # that changes, this might become invalid, so I added a warning in case the 
   # directory does not exist.
-  (my $subdir = "subdir.${probename}_$soaphost") =~ s|[:/]|_|g;
+  (my $subdir = "${probename}_$soaphost") =~ s|[:/]|_|g;
   $subdir = "$workingfolder/gratiafiles/$subdir";
   if(!-e $subdir) {
       setWarning();
